### ISSUE TITLE
[chat] get out of chat by hitting "escape" key

### ISSUE THREAD

**Issue by msouth** [ironbane/IronbaneServerLegacy#390]
It would be nice if hitting "escape" in the chat would close the chat.

On more than one occasion I've accidentally opened the chat in a battle, or opened it to say something quick.  It would be nice to have a quick way to undo that.

**Comment by eric-ycw** [comment]
The esc key is already used for "Back to the main menu"

**Comment by msouth** [comment]
Yes, but that function is not active when the chat is active.  If you hit esc when you are in chat, nothing happens.

In the Android UI, there is a sort of "contextual back" (which is mapped, on my phone, to a button at the bottom of the phone) that can mean "back to the last part of the interface" or "back to the home screen" or "back to the app that opened this app".  You don't lose anything (I would argue) by allowing esc in chat to take you back to the last part of the app, and esc in normal mode to give you the "back to main" prompt.

Even a user who didn't understand this would probably think it just didn't take and hit esc again, and it would then do what they would expect, so it would be sort of self-teaching.

While we're on the topic, it would be nice if hitting esc when the 'back to main' dialog is up would close the 'back to main' dialog.  I'll put that as a separate issue.  [edit:  done  #393 ]

**Comment by ghost** [comment]
Again not trying to steal other game ideas buuuuut:
function on.key(escape)
if (focus.chat == true)
focus.chat = false;
else open(menu);

This is how most other MMOs handle escape keys. If focus is on any part of 
the GUI, stop focusing there. Otherwise, just bring up the menu.

Might lead to some shenanigans, but other games don't seem to gave an issue.

Sent with AquaMail for Android
http://www.aqua-mail.com

**Comment by ghost** [comment]
@msouth that was just some quick psuedocode to get my point across. Obviously there are flags and all sorts of other such ways this can be accomplished, and since it's a popular mechanic used in other MMOs, it shouldn't confuse anyone. May need to be part of the "help" dialogs but should not make for any major player issues.
